src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -94,34 +94,19 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
             dbsApi.insertBulkBlock(blockDump=block)
             results.put({'name': name, 'success': "uploaded"})
         except HTTPError as ex:
-            # when we will have new dbs3-client and fine granularity DBS Go server
-            # error codes then we should remove if/elif/else and replace it with
-            # DBSError(ex.body), e.g.
-            # dbsError = DBSError(ex.body)
-            # reason = dbsError.getReason()
-            # srvCode = dbsError.getServverCode()
-            # results.put({'name': name, 'success': 'error', 'error': reason, 'server_code': srvCode})
-
-            # legacy code which deal with DBS python server exceptions
             dbsError = DBSError(ex.body)
             reason = dbsError.getReason()
-            if 'Block %s already exists' % name in reason:
-                # Then this is probably a duplicate
-                # Ignore this for now
+            message = dbsError.getMessage()
+            srvCode = dbsError.getServerCode()
+            msg = f'DBSError: {message}, reason: {reason}'
+            if srvCode == 128:
+                # block already exist
                 logging.warning("Block %s already exists. Marking it as uploaded.", name)
                 logging.debug("Exception: %s", reason)
                 results.put({'name': name, 'success': "uploaded"})
-            # python server error produces Missing message, while
-            # Go based server produces unable to find dataset_id message
-            elif 'Missing data when inserting to dataset_parents' in reason or \
-                'unable to find dataset_id' in reason:
-                msg = "Parent dataset is not inserted yet for block %s." % name
+            elif srvCode in [129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140]:
```

Comment:
From:
https://github.com/dmwm/dbs2go/blob/000d8b90c88bdf2faf0128f6f5371671d0762389/dbs/errors.go
it looks like code 140 is the one we can face when injecting the very first block of a dataset (where the block insertion is done before the dataset itself), right?

If so, I would then change this elif block to (giving you the whole code here):
```
elif srvCode == 140:
    logging.warning("Hit a transient data race condition injecting block %s", name)
    results.put({'name': name, 'success': "error", 'error': msg})
else:
    msg = f"Error trying to process block {name} through DBS. Details: {msg}"
    logging.error(msg)
    results.put({'name': name, 'success': "error", 'error': msg})
```

In plain english:
* whenever it's a known and transient error, we record a warning log record;
* and when it's another problem that will keep failing until action is taken, an error log record is created.

Reply:
done, the transient error (racing condition) happens on many error codes, e.g. primary or processed datasets should exist, data tier should exist, etc. Therefore, I adjusted code accordingly.